io/net/net_data.py

```diff
@@ -161,10 +169,10 @@ def interface_wait(self, cmd):
         '''
          Waits for the interface to come up
         '''
-        for i in range(0, 600, 5):
+        for _ in range(0, 600, 5):
```

Comment:
We have a `wait()` in our utils. This method could be changed to a method which checks once and returns `True`/`False`, then it's called by our `wait.wait()`. Example:
```python
from avocado.utils import wait
...

    def setUp(self):
        self.if_wait_cmd = None
        ....

    def is_interface_ready(self):
        if self.if_wait_cmd is None:
            return False
        try:
            output = process.system_output(self.if_wait_cmd, shell=True)
        except process.CmdError:
            return False

        if 'state UP' or 'Link detected: yes' in output:
            self.log.info("%s is ready", self.interface)
            return True

        return False

    def testinterface(self):
        ...
        self.if_wait_cmd = ip_link
        self.assertTrue(wait.wait_for(is_interface_ready, timeout=60, 'Interface test failed.')
        ...
        self.if_wait_cmd = "ip link show %s | head -1" % self.interface
        self.assertTrue(wait.wait_for(is_interface_ready, timeout=60, 'Interface test failed.')
```
Code above is not tested, but it demonstrates what I mean.